src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp

```diff
@@ -19,7 +19,7 @@ namespace rdf4cpp::rdf::datatypes::registry::util {
  * @return the resulting value
  * @throws std::runtime_error if the string cannot be parsed
  */
-template<std::integral I>
+template<std::integral I, ConstexprString datatype>
```

Comment:
`ConstexprString datatype` does not enforce in any way to be set to correct values only. It's very easy to set it to some random value. I have no simple idea at hand but a more robust approach would be preferable. This does not need to be addressed here but can be moved to a separate issue.

Reply:
moved to error message issue